pkg/codegen/python/importer.go

```diff
@@ -98,3 +98,22 @@ func (importer) ImportPackageSpec(pkg *schema.Package, raw json.RawMessage) (int
 	}
 	return info, nil
 }
+
+// minimumPythonVersion returns a string containing the version
+// constraint specifying the minimumal version of Python required
+// by this package. For example, ">=3.8" is satified by all versions
+// of Python greater than or equal to Python 3.8.
+func (info PackageInfo) minimumPythonVersion() (string, error) {
+	if info.PythonRequires != "" {
+		return info.PythonRequires, nil
+	}
+	return "", noMinimumPythonErr{}
+}
+
+// noMinimumPythonErr is a non-fatal error indicating that the schema
+// did not provide a minimum version of Python for the Package.
+type noMinimumPythonErr struct{}
```

Comment:
If we do keep the method and error object:

The error message for this is constant.
Could we just use a standard sentinel error here instead?

```go
var errNoMinimumPython = errors.New(..)
```

Reply:
I typically prefer to use the newtype pattern so that the error is introspectable with a type switch. I like it for future-proofing, in case down the road we need to differentiate between this error, which is nonfatal, and another error which could be fatal.

If that's an insufficient explanation, I'm happy to use `errors.New()` instead; I just find that approach to be less adaptable.

Reply:
Fair, I can see that being a need from The Before Times, but I don't think this is necessary since Go 1.13.
Type switches are no longer the recommended way to match errors because they break with error wrapping (whether with %w or a custom error type wrapper).

errors.Is and errors.As are what we'd want to use for error matching (depending on sentinel error value or custom error type), and errors.New works just fine with errors.Is.

Reply:
Thanks for the advice here. I went back and made the changes:
1. Moved the validation code into a top-level function colocated with metadata generation, instead of using a method on `PackageInfo`.
2. Switched to using `errors.New` instead of an error struct.